Frameworks/jdi-matchers/src/main/java/com/epam/matcher/base/IChecker.java

```diff
@@ -13,128 +13,588 @@
+     * @param <T>
+     */
     <T> void listEquals(Collection<T> actual, Collection<T> expected, String failMessage);
+
+    /**
+     * Asserts if two collections of {@link T} are equals. If not, an AssertionError is thrown.
+     *
+     * @param actual   actual collection of T
+     * @param expected expected collection of T
+     * @param <T>
+     */
     <T> void listEquals(Collection<T> actual, Collection<T> expected);
+
+    /**
+     * Asserts that actual collection contains expected. If not, an AssertionError,
+     * with the given message, is thrown.
+     *
+     * @param actual      actual collection of T
+     * @param expected    expected collection of T
+     * @param failMessage the assertion error message
+     * @param <T>
+     */
     <T> void listContains(Collection<T> actual, Collection<T> expected, String failMessage);
+
+    /**
+     * Asserts that actual collection contains expected. If not, an AssertionError is thrown.
+     *
+     * @param actual   actual collection of T
+     * @param expected expected collection of T
+     * @param <T>
+     */
     <T> void listContains(Collection<T> actual, Collection<T> expected);
+
+    /**
+     * Asserts that two arrays are equals. If not, an AssertionError,
+     * with the given message, is thrown.
+     *
+     * @param actual      actual array
+     * @param expected    expected array
+     * @param failMessage the assertion error message
+     * @param <T>
+     */
     <T> void arrayEquals(T[] actual, T[] expected, String failMessage);
+
+    /**
+     * Asserts that two arrays are equals. If not, an AssertionError is thrown.
+     *
+     * @param actual   actual array
+     * @param expected expected array
+     * @param <T>
+     */
     <T> void arrayEquals(T[] actual, T[] expected);
     // endregion
 
     // TODO entityIncludeMapArray
 
     // region Sort Array Integer
+
+    /**
+     * Check that array is sorted ascending. If not, an AssertionError,
+     * with the given message, is thrown.
+     *
+     * @param array       array to check sorting order
+     * @param failMessage the assertion error message
+     */
     void isSortedByAsc(int[] array, String failMessage);
+
+    /**
+     * Check that array is sorted ascending. If not, an AssertionError is thrown.
+     *
+     * @param array array to check sorting order
+     */
     void isSortedByAsc(int[] array);
+
+    /**
+     * Check that array is sorted descending. If not, an AssertionError,
+     * with the given message, is thrown.
+     *
+     * @param array       array to check sorting order
+     * @param failMessage the assertion error message
+     */
     void isSortedByDesc(int[] array, String failMessage);
+
+    /**
+     * Check that array is sorted descending. If not, an AssertionError is thrown.
+     *
+     * @param array array to check sorting order
+     */
     void isSortedByDesc(int[] array);
     // endregion
 
-    // ListProcessor
+    // ListProcessor todo: javadoc
```

Comment:
ToDo? =)

Reply:
Firstly was added in implementation. Now duplicated to interface.